packages/platforms/dni/x86-64/x86-64-dni-wb2448/onlp/builds/src/module/src/fani.c

```diff
@@ -0,0 +1,156 @@
+/************************************************************
+ * <bsn.cl fy=2014 v=onl>
+ *
+ *        Copyright 2014, 2015 Big Switch Networks, Inc.
+ *        Copyright 2017 Delta Networks, Inc.
+ *
+ * Licensed under the Eclipse Public License, Version 1.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *        http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *
+ * </bsn.cl>
+ ************************************************************
+ *
+ * Fan Platform Implementation Defaults.
+ *
+ ***********************************************************/
+#include <onlp/platformi/fani.h>
+#include "x86_64_dni_wb2448_int.h"
+
+#include <sys/mman.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <onlplib/mmap.h>
+
+#include "platform_lib.h"
+
+#define VALIDATE(_id)                       \
+do 											\
+{	                                        \
+    if(!ONLP_OID_IS_FAN(_id)) 				\
+	{        								\
+        return ONLP_STATUS_E_INVALID;       \
+    }                                       \
+} while(0)
+		
+#define MAX_FAN_SPEED 18000
```

Comment:
It would be better if this were added as a CONFIG_ define in the module's YAML file.